builtin/providers/aws/resource_aws_kinesis_firehose_delivery_stream.go

```diff
@@ -91,49 +186,179 @@ func resourceAwsKinesisFirehoseDeliveryStream() *schema.Resource {
 	}
 }
 
+func validateConfiguration(d *schema.ResourceData) error {
+	destination := d.Get("destination").(string)
+	if destination != "s3" && destination != "redshift" {
+		return fmt.Errorf("[ERROR] Destination must be s3 or redshift")
+	}
+
+	s3Configuration := d.Get("s3_configuration").([]interface{})
+	if len(s3Configuration) > 1 {
+		return fmt.Errorf("[ERROR] You can only define a single s3_configuration per delivery stream")
```

Comment:
We no longer need to do this, we can add `MaxItems: 1` to the schema for `s3_configuration` :)